statusHandler/presenter/instanceInfoGetters.go

```diff
@@ -41,3 +51,83 @@ func (psh *PresenterStatusHandler) GetCountLeader() uint64 {
 func (psh *PresenterStatusHandler) GetCountAcceptedBlocks() uint64 {
 	return psh.getFromCacheAsUint64(core.MetricCountAcceptedBlocks)
 }
+
+// GetNodeName will return node name
+func (psh *PresenterStatusHandler) GetNodeName() string {
+	nodeName := psh.getFromCacheAsString(core.MetricNodeDisplayName)
+	if nodeName == "" {
+		nodeName = "noname"
+	}
+
+	return nodeName
+}
+
+// GetTotalRewardsValue will return total value of rewards and how rewards was increased on every second
+// Rewards estimation will be equal with :
+// (numSignedBlocks - numProposedBlocks) * communityPercentage * Rewards +
+// numProposedBlocks * leaderPercentage * Rewards
+func (psh *PresenterStatusHandler) GetTotalRewardsValue() (string, string) {
+	rewardsValue := psh.getBigIntFromStringMetric(core.MetricRewardsValue)
+	communityPercentage := psh.getFloatFromStringMetric(core.MetricCommunityPercentage)
+	leaderPercentage := psh.getFloatFromStringMetric(core.MetricLeaderPercentage)
+	numSignedBlocks := psh.getFromCacheAsUint64(core.MetricCountConsensusAcceptedBlocks)
+	numProposedBlocks := psh.getFromCacheAsUint64(core.MetricCountAcceptedBlocks)
+
+	rewardsCommunityCoefficient := float64(numSignedBlocks-numProposedBlocks) * communityPercentage
```

Comment:
you should be careful when number of proposed blocks is greater than number of signed blocks. what happens then ?

Reply:
Or you can do the same way as the method above and place a "0" if you find inital values

Reply:
Done. I'm using modulo now because at the beginning it is possible that the number of proposed blocks to be greater than number of signed and accepted blocks.